### PR TITLE
Disabled dropdowns are now greyed out TRNG-1183

### PR DESCRIPTION
**Pull request by Gusinuhe**
### Description

**Fixes**: An issue where disabled dropdowns were still white giving the impression that they can be selected.

image

image

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Runtime/CourseController/AdvancedCourseController.cs [hunk 1]
```diff
@@ -458,7 +458,13 @@ private void SetupLanguagePicker()
             });
             
             // If there is only one option, the dropdown is currently disabled.
-            languagePicker.enabled = supportedLanguages.Count > 1;
+            if (supportedLanguages.Count > 1 == false)
```
Comment (Maxsiom) [inline Runtime/CourseController/AdvancedCourseController.cs:461]:
```suggestion
            if (supportedLanguages.Count < 2)
```
Seems more readable to me. But I am also fine without the change.

File: Runtime/CourseController/AdvancedCourseController.cs [hunk 2]
```diff
@@ -479,7 +485,13 @@ private void SetupModePicker()
             modePicker.value = RuntimeConfigurator.Configuration.Modes.CurrentModeIndex;
             
             // If there is only one option, the dropdown is currently disabled.
-            modePicker.enabled = availableModes.Count > 1;
+            if (availableModes.Count > 1 == false)
```
Comment (Maxsiom) [inline Runtime/CourseController/AdvancedCourseController.cs:488]:
```suggestion
            if (availableModes.Count < 2)
```
Seems more readable to me. But I am also fine without the change.

File: Runtime/CourseController/AdvancedCourseController.cs [hunk 3]
```diff
@@ -520,15 +532,23 @@ private void PopulateChapterPickerOptions(int startingIndex)
...
                 chapterPicker.AddOptions(dropdownOptions);
-
+                
                 // Reset the selected value
                 chapterPicker.value = 0;
-
+                
                 // If there is only one option, the dropdown is currently disabled.
                 chapterPicker.enabled = dropdownOptions.Count > 1;
+                if (dropdownOptions.Count > 1 == false)
```
Comment (Maxsiom) [inline Runtime/CourseController/AdvancedCourseController.cs:544]:
```suggestion
                if (dropdownOptions.Count < 2)
```
Seems more readable to me. But I am also fine without the change.

File: Runtime/CourseController/StandaloneCourseController.cs [hunk 4]
```diff
@@ -459,7 +459,13 @@ private void SetupLanguagePicker()
             });
             
             // If there is only one option, the dropdown is currently disabled.
-            languagePicker.enabled = supportedLanguages.Count > 1;
+            if (supportedLanguages.Count > 1 == false)
```
Comment (Maxsiom) [inline Runtime/CourseController/StandaloneCourseController.cs:462]:
```suggestion
            if (supportedLanguages.Count < 2)
```
Seems more readable to me. But I am also fine without the change.

File: Runtime/CourseController/StandaloneCourseController.cs [hunk 5]
```diff
@@ -480,7 +486,13 @@ private void SetupModePicker()
             modePicker.value = RuntimeConfigurator.Configuration.Modes.CurrentModeIndex;
             
             // If there is only one option, the dropdown is currently disabled.
-            modePicker.enabled = availableModes.Count > 1;
+            if (availableModes.Count > 1 == false)
```
Comment (Maxsiom) [inline Runtime/CourseController/StandaloneCourseController.cs:489]:
```suggestion
            if (availableModes.Count < 2)
```
Seems more readable to me. But I am also fine without the change.

File: Runtime/CourseController/StandaloneCourseController.cs [hunk 6]
```diff
@@ -529,7 +541,14 @@ private void PopulateChapterPickerOptions(int startingIndex)
                 chapterPicker.value = 0;
 
                 // If there is only one option, the dropdown is currently disabled.
-                chapterPicker.enabled = dropdownOptions.Count > 1;
+                if (dropdownOptions.Count > 1 == false)
```
Comment (Maxsiom) [inline Runtime/CourseController/StandaloneCourseController.cs:544]:
```suggestion
                if (dropdownOptions.Count < 2)
```
Seems more readable to me. But I am also fine without the change.

File: Runtime/CourseController/AdvancedCourseController.cs (hunk 1 above)
Comment (SimonTheSourcerer) [inline Runtime/CourseController/AdvancedCourseController.cs:461]:
```suggestion
            if (supportedLanguages.Count <= 1)
```

**Review by Maxsiom — APPROVED** [review]
Looks good. And works for me on Windows.

File: Runtime/CourseController/AdvancedCourseController.cs (hunk 2 above)
Comment (SimonTheSourcerer) [inline Runtime/CourseController/AdvancedCourseController.cs:488]:
```suggestion
            if (availableModes.Count <= 1)
```

File: Runtime/CourseController/AdvancedCourseController.cs (hunk 3 above)
Comment (SimonTheSourcerer) [inline Runtime/CourseController/AdvancedCourseController.cs:544]:
```suggestion
                if (dropdownOptions.Count <= 1)
```

File: Runtime/CourseController/StandaloneCourseController.cs (hunk 4 above)
Comment (SimonTheSourcerer) [inline Runtime/CourseController/StandaloneCourseController.cs:462]:
```suggestion
            if (supportedLanguages.Count <= 1)
```

File: Runtime/CourseController/StandaloneCourseController.cs (hunk 5 above)
Comment (SimonTheSourcerer) [inline Runtime/CourseController/StandaloneCourseController.cs:489]:
```suggestion
            if (availableModes.Count <= 1)
```

File: Runtime/CourseController/StandaloneCourseController.cs (hunk 6 above)
Comment (SimonTheSourcerer) [inline Runtime/CourseController/StandaloneCourseController.cs:544]:
```suggestion
                if (dropdownOptions.Count <= 1)
```

**Review by SimonTheSourcerer — DISMISSED** [review]

**Comment by SimonTheSourcerer** [comment]
I dont own a quest, so I cannot completly test this, feel free to dismiss me after applying changes.

**Comment by Gusinuhe** [comment]
I'll leave this open a little longer hoping that someone with a Quest could test it.

**Comment by tomwim** [comment]
> I'll leave this open a little longer hoping that someone with a Quest could test it.

@Luphoxa pleeeeease :)

**Review by SimonTheSourcerer — APPROVED** [review]
only tested on windows, but it works

**Pull request merged**